core/src/main/scala/kafka/tools/EndToEndLatency.scala

```diff
@@ -83,21 +83,32 @@ object EndToEndLatency {
     val producer = new KafkaProducer[Array[Byte], Array[Byte]](producerProps)
 
     // sends a dummy message to create the topic if it doesn't exist
-    producer.send(new ProducerRecord[Array[Byte], Array[Byte]](topic, Array[Byte]())).get()
+    val dummyRecord = producer.send(new ProducerRecord[Array[Byte], Array[Byte]](topic, Array[Byte]())).get()
```

Comment:
Any reason not to just switch to AdminClient and validate topic creation before proceeding? It seems like this wasn't mentioned in the previous discussion on the PR that introduced this dummy message, but maybe I scanned over it too quickly.

I think we may have done it this way to preserve the functionality in that we wouldn't *fail* on a cluster with topic auto creation, but arguably the old implementation pre-dummyRecord was simply buggy since at least the first message would incur the topic creation overhead and skew results. Main reason for being careful about changing it is that failing fast would result in an error exit status rather than just skewed results. On the other hand, this code doesn't even work with a topic without auto topic creation on, even if the principal has topic creation permissions, so I'm not sure this is really the right behavior either.

@hachikuji, looks like you reviewed the previous update. Thoughts?

Reply:
Yeah, using the `AdminClient` seems preferable over relying on auto topic creation. One disadvantage is that clusters older than 0.10.1 would no longer be supported unless the topic was created manually. But this seems OK since creating the topic manually is not a big deal if someone was trying to compare the performance of very old clusters.

Reply:
There was no reason not to use AdminClient, except trying to preserve the functionality as closely as possible. However, after reading comments on the previous PR (5319), it seems like the behavior was not really well defined. So, I agree with using AdminClient for topic creation. I changed the code to check if topic exists and use AdminClient to create the topic.